azplugins/flow.py

```diff
@@ -746,13 +767,51 @@ def density(self):
             return np.zeros(self.bins)
 
     @property
-    def velocity(self):
-        r"""The current average velocity profile."""
+    def mass_density(self):
+        r"""The current mass-averaged density profile."""
+        if hoomd.comm.get_rank() != 0:
+            hoomd.context.msg.error('Flow profile only defined on root rank.\n')
+            raise RuntimeError('Flow profile only defined on root rank')
+
+        if self.samples > 0:
+            return np.divide(self._bin_mass, self._counts, out=np.zeros(self.bins), where=self._counts > 0)
+        else:
+            return np.zeros(self.bins)
+
+    @property
+    def number_velocity(self):
+        r"""The current number-averaged velocity profile ."""
+        if hoomd.comm.get_rank() != 0:
+            hoomd.context.msg.error('Flow profile only defined on root rank.\n')
+            raise RuntimeError('Flow profile only defined on root rank')
+
+        result = np.zeros_like(self._number_velocity)
+        if self.samples > 0:
+            for dim in range(3):
+               result[dim] = self._number_velocity[dim]/self.samples
```

Comment:
Can't you just do:
```
if self.samples > 0:
    return self._number_velocity / self.samples
else:
    return np.zeros(self.bins)
```
?

Reply:
yes, absolutely. This is simply leftover copy-pasted code from prior versions.